components/expression/expression-loader.js

```diff
@@ -52,117 +54,237 @@ genotet.ExpressionLoader.prototype.loadExpressionMatrix_ = function(matrixName,
   };
 
   $.get(genotet.data.serverURL, params, function(data) {
-    // Store the last applied data selectors.
-    _(data).extend({
-      matrixname: matrixName,
-      geneRegex: geneRegex,
-      conditionRegex: conditionRegex
-    });
+      // Store the last applied data selectors.
+      _(data).extend({
+        matrixname: matrixName,
+        geneRegex: geneRegex,
+        conditionRegex: conditionRegex
+      });
 
-    this.data.matrix = data;
+      this.data.matrix = data;
 
-    this.signal('loadComplete');
-  }.bind(this), 'jsonp')
+      this.signal('loadComplete');
+    }.bind(this), 'jsonp')
     .fail(this.fail.bind(this, 'cannot load expression matrix', params));
 };
 
-
-/*
-LoaderHeatmap.prototype.updateData = function(identifier) {
-  if (identifier.action == 'node') {
-    this.loadLine(this.lastIdentifier.mat, identifier.name);
-    if (identifier.net != null) {
-      this.loadHeatmapTargets(identifier.net, identifier.name);
-    }
-  }else if (identifier.action == 'link') {
-    this.clearLines();
-    this.loadLine(this.lastIdentifier.mat, identifier.source);
-    this.loadLine(this.lastIdentifier.mat, identifier.target);
+/**
+ * Updates the genes in the current expression.
+ * @param {string} method Update method, either 'set' or 'add'.
+ * @param {string} geneRegex Regex that selects the genes to be updated.
+ */
+genotet.ExpressionLoader.prototype.updateGenes = function(method, geneRegex, originParams) {
+  var regex = this.data.matrix.geneRegex;
+  geneRegex = geneRegex.toUpperCase();
+  switch (method) {
+    case 'setGene':
+      // Totally replace the regex.
+      regex = geneRegex;
+      this.data.genes = [];
+      this.data.genes.push(geneRegex);
```

Comment:
What exactly is genes?  Is it an array of gene names? But you are pushing regex into the array and this makes me confusing. What happenes if geneRegex = "aaa|bbb|ccc" or even "^[a-z]+[0-9]*"?